analysis/src/SignatureHelp.ml

```diff
@@ -296,19 +306,24 @@ let signatureHelp ~path ~pos ~currentFile ~debug =
              in the form of a list of start/end character offsets. We leverage the parser to figure the offsets out by parsing the label, and extract the
              offsets from the parser. *)
 
-          (* Put together a label here that both makes sense to show to the end user in the signature help, but also can be passed to the parser. *)
-          let label = "let " ^ name ^ ": " ^ Shared.typeToString type_expr in
+          (* A full let binding with the type text is needed for the parser to be able to parse it.  *)
+          let labelPrefix = "let fn: " in
+          let labelPrefixLen = String.length labelPrefix in
+          let fnTypeStr = Shared.typeToString type_expr in
+          let typeStrForParser = labelPrefix ^ fnTypeStr in
           let {Res_driver.parsetree = signature} =
             Res_driver.parseInterfaceFromSource ~forPrinter:false
-              ~displayFilename:"<missing-file>" ~source:label
+              ~displayFilename:"<missing-file>" ~source:typeStrForParser
           in
 
-          let parameters = extractParameters ~signature ~label in
+          let parameters =
+            extractParameters ~signature ~typeStrForParser ~labelPrefixLen
+          in
           if debug then
             Printf.printf "extracted params: \n%s\n"
               (parameters
               |> List.map (fun (_, start, end_) ->
-                     String.sub label start (end_ - start))
+                     String.sub fnTypeStr start (end_ - start))
```

Comment:
Do we know these are safe bounds for the substring?

Reply:
Yes, they are derived from the locations the parser tells us so they should be fine. This also only runs in the tests so an exception shouldn't be too damaging.